### PR TITLE
feat(contact_form): Enable selection of issue type and enable Jira integration

### PR DESCRIPTION
**Pull request by elisa-a-v**
Addresses #4585 

This update enhances the contact form by allowing users to select the "Issue Type" for their submission, and also optionally pre-select it via URL query parameters, improving UX.

Additionally, the contact form now sends submissions to `support@freelawproject.atlassian.net`, integrating with Jira for better issue tracking. The email body includes:

- User Email: Displayed as `User Email: <email address>` to facilitate easy parsing in Jira.
- Issue Type: Clearly stated to assist in automatic categorization within Jira.
- Subject and Message: Retained as before for detailed context.

This change enables automatic ticket creation to relieve some of the overhead of managing requests manually by email.

### CODE REVIEW

**Comment by elisa-a-v** [comment]
@mlissner I had issues with the pre-commit when working on this which is why there are so many commits each doing very little, but I got it working now so it won't happen again!

I can revert them and add all the small changes in a single commit, please let me know if I should do it or just leave it as is.

**Review by mlissner — APPROVED** [review]
Looks good to me at a glance. Off to Eduardo for more careful review. The commits are fine, we can just squash when we merge.

Thank you and congrats on your first (official) PR!

**Comment by mlissner** [comment]
@s-taube, before we merge this, have you had a chance to look at the security vuln I mentioned that this might open up?

File: cl/simple_pages/forms.py [hunk 1]
```diff
@@ -6,6 +6,12 @@
 
 
 class ContactForm(forms.Form):
+    ISSUE_TYPE_CHOICES = [
+        ("removal", "Case Removal Request"),
+        ("bug", "RECAP Extension Bug"),
+        ("support", "Support"),
+    ]
+
```
Comment (ERosendo) [inline cl/simple_pages/forms.py:14]: We should adopt Django's recommendation of using named constants for each value of the iterable.

```suggestion
    REMOVAL_REQUEST = "removal"
    RECAP_BUG = "recap"
    SUPPORT_REQUEST = "support"
    ISSUE_TYPE_CHOICES = [
        (REMOVAL_REQUEST, "Case Removal Request"),
        (RECAP_BUG, "RECAP Extension Bug"),
        (ISSUE_TYPE_CHOICES, "Support"),
    ]
   VALID_ISSUE_TYPES = [REMOVAL_REQUEST, RECAP_BUG, SUPPORT_REQUEST]
```

By following this suggestion, we can write cleaner and more expressive checks, such as:

```python
if form.issue_type == ContactForm.REMOVAL_REQUEST
```

and 

```python
if issue_type in ContactForm.VALID_ISSUE_TYPES:
```

File: cl/simple_pages/views.py [hunk 2]
```diff
@@ -378,24 +378,37 @@ async def contact(
...
     else:
         # the form is loading for the first time
+        issue_type = request.GET.get("issue_type")
+        if issue_type:
+            issue_type_lower = issue_type.lower()
+            valid_issue_types = [
+                choice[0] for choice in ContactForm.ISSUE_TYPE_CHOICES
+            ]
+            if issue_type_lower in valid_issue_types:
+                initial["issue_type"] = issue_type_lower
```
Comment (ERosendo) [inline cl/simple_pages/views.py:411]: if we apply the previous suggestion, we can refactor this code like this:

```suggestion
        if issue_type and issue_type.lower() in ContactForm.VALID_ISSUE_TYPES:
            initial["issue_type"] = issue_type.lower()
```

**Comment by s-taube** [comment]
> @s-taube, before we merge this, have you had a chance to look at the security vuln I mentioned that this might open up?

Yes. As far as I can tell, this Zendesk issue should not impact Atlassian products at all.

**Review by ERosendo — CHANGES_REQUESTED** [review]
The code is looking great! Just a couple of small things to address before we can merge this PR.

**Comment by ERosendo** [comment]
LGTM :shipit:

**Comment by mlissner** [comment]
OK, here we go. Thanks everybody. @s-taube, in about ten minutes, when [this](https://github.com/freelawproject/courtlistener/actions/runs/11464656156) finishes, this code will be live and I think we'll be in Jira-land.

**Comment by mlissner** [comment]
Thanks all!

**Pull request merged**